bookkeeper-server/src/test/java/org/apache/bookkeeper/replication/AuditorLedgerCheckerTest.java

```diff
@@ -321,6 +325,120 @@ public void testReadOnlyBookieShutdown() throws Exception {
+        lh1.close();
+
+        final CountDownLatch underReplicaLatch = registerUrLedgerWatcher(ledgerList
+                .size());
+
+        // wait for 5 seconds before starting the recovery work when a bookie fails
+        baseConf.setLostBookieRecoveryDelay(5);
+
+        // shutdown a non auditor bookie; choosing non-auditor to avoid another election
+        String shutdownBookie = shutDownNonAuditorBookie();
+
+        LOG.debug("Waiting for ledgers to be marked as under replicated");
+        assertFalse("audit of lost bookie isn't delayed", underReplicaLatch.await(4, TimeUnit.SECONDS));
+        assertEquals("under replicated ledgers identified when it was not expected", 0,
+                urLedgerList.size());
+
+        // wait for another 5 seconds for the ledger to get reported as under replicated
+        assertTrue("audit of lost bookie isn't delayed", underReplicaLatch.await(2, TimeUnit.SECONDS));
+
+        assertTrue("Ledger is not marked as underreplicated:" + ledgerId,
+                urLedgerList.contains(ledgerId));
+        Map<Long, String> urLedgerData = getUrLedgerData(urLedgerList);
+        String data = urLedgerData.get(ledgerId);
+        assertTrue("Bookie " + shutdownBookie
+                + "is not listed in the ledger as missing replica :" + data,
+                data.contains(shutdownBookie));
+    }
+
+    /**
+     * Test publishing of under replicated ledgers by the auditor
+     * bookie is delayed if LostBookieRecoveryDelay option is set
+     */
+    @Test(timeout=60000)
+    public void testDelayedAuditOfLostBookies() throws Exception {
+        _testDelayedAuditOfLostBookies();
+    }
+
+    /**
+     * Test publishing of under replicated ledgers by the auditor
+     * bookie is delayed if LostBookieRecoveryDelay option is set
+     * and it continues to be delayed even when periodic bookie check
+     *  is set to run every 2 secs. I.e. periodic bookie check doesn't
+     *  override the delay
+     */
+    @Test(timeout=60000)
+    public void testDelayedAuditWithPeriodicBookieCheck() throws Exception {
+        // enable periodic bookie check on a cadence of every 2 seconds.
+        // this requires us to stop the auditor/auditorElectors, set the
+        // periodic check interval and restart the auditorElectors
+        stopAuditorElectors();
+        baseConf.setAuditorPeriodicBookieCheckInterval(2);
+        startAuditorElectors();
+
+        // wait for half a second so that the initial periodic check finishes
+        Thread.sleep(500);
+
+        // the delaying of audit should just work despite the fact
+        // we have enabled periodic bookie check
+        _testDelayedAuditOfLostBookies();
+    }
+
+    /**
+     * Test audit of bookies is delayed when one bookie is down. But when
+     * another one goes down, the audit is started immediately.
+     */
+    @Test(timeout=60000)
+    public void testDelayedAuditWithMultipleBookieFailures() throws Exception {
```

Comment:
I added some delay before starting testDelayedAuditWithMultipleBookieFailures() so that the initial bookie check finishes. If the system is busy the initial check can be very slow and could end up doing the audit of lost bookies. @eolivelli, hopefully this should address the test failure on your setup.